admin-frontend/src/components/youthWorker.js

```diff
@@ -16,17 +16,33 @@ import {
 } from 'react-admin';
 import { getYouthClubs } from '../utils';
 
-export const YouthWorkerList = (props) => (
-  <List title="Nuorisotyöntekijät" bulkActionButtons={false} exporter={false} pagination={false} {...props}>
-    <Datagrid>
-      <FunctionField label="Nimi" render={record => `${record.firstName} ${record.lastName}`} />
-      <TextField label="Sähköposti" source="email" />
-      <TextField label="Kotinuorisotila" source="mainYouthClub" />
-      <BooleanField label="Ylläpitäjä" source="isSuperUser" />
-      <EditButton />
-    </Datagrid>
-  </List>
-);
+export const YouthWorkerList = (props) => {
+  const [youthClubs, setYouthClubs] = useState([]);
+  useEffect(() => {
+    const addYouthClubsToState = async () => {
+      const parsedYouthClubs = await getYouthClubs();
+      setYouthClubs(parsedYouthClubs);
+    };
+    addYouthClubsToState();
+  }, []);
+
+  const ycIdToName = (id) => {
+    const match = youthClubs.find(yc => { return yc.id === id });
+    return match ? match.name : "?";
+  }
+
+  return (
+    <List title="Nuorisotyöntekijät" bulkActionButtons={false} exporter={false} pagination={false} {...props}>
+      <Datagrid>
+        <FunctionField label="Nimi" render={record => `${record.firstName} ${record.lastName}`} />
+        <TextField label="Sähköposti" source="email" />
+        <FunctionField label="Kotinuorisotila" render={record => `${ycIdToName(record.mainYouthClub)}`} />
```

Comment:
Actually this ycIdToName manual check is unnecessary, <SelectField label="Kotinuorisotalo" source="mainYouthClub" choices={youthClubs}/> would work just fine by itself  :)

Reply:
Thanks, somehow I missed this. Thought I tried it out but apparently not. Merging like this.